File: leetcode/0667-beautiful-arrangement-ii/constructArray.go
package leetcode

func constructArray(n int, k int) []int {
	var result = []int{}

	var limit = n - k
	for i := 1; i <= limit; i++ {
		result = append(result, i)
	}

	for max, min := n, limit+1; min <= max; max, min = max-1, min+1 {
		result = append(result, max)

		if max == min {
			continue
		}
		result = append(result, min)
	}

	return result
}

File: leetcode/0667-beautiful-arrangement-ii/constructArray.ts
function constructArray(n: number, k: number): number[] {
  var limit = n - k;
  var result = [];
  for (let i = 1; i <= n; i++) {
    if (i <= limit) {
      result.push(i);
      continue;
    }
  }
  for (let min = limit + 1, max = n; min <= max; min++, max--) {
    result.push(max);
    if (max == min) {
      continue;
    }
    result.push(min);
  }

  return result;
}

export default constructArray;
